scapy/packet.py

```diff
@@ -1005,6 +1005,9 @@ def getlayer(self, cls, nb=1, _track=None, _subclass=False, **flt):
         """
         if _subclass:
             match = lambda cls1, cls2: issubclass(cls1, cls2)
+        elif isinstance(self, _PacketMatchSubclass):
+            match = lambda cls1, cls2: cls1 == cls2 or \
```

Comment:
I'm not sure to understand: how is that different from `issubclass(cls1, cls2)?

Reply:
It ensures direct inheritance
![195507a4-8f24-404f-bfdf-29e3b274e461](https://user-images.githubusercontent.com/10530980/47574425-ddabf000-d93f-11e8-95fd-1a04177bc5a2.jpeg)

Reply:
And do we need it?

Reply:
With
```
a = Ether()/Dot11FCS()
```
We want `a[Dot11]` to work, but not `a[Packet]`

Reply:
I do not really agree, but maybe I don't understand the implications here.

I think it's fine if `a[Packet]` works, and it makes the code easier as it relies on standard Python inspection functions.

Reply:
Well I think it would really get tricky, we’ll have
```
(Ether()/Dot11FCS()/IP())[Packet]
```
Returning `Dot11FCS` for no apparent reason.

Also a lot of people generally fail to use `[]` properly (I see a lot of `pkt[0]` or `pkt["IP"]`) so I would prefer to have its purpose precisely defined (i’d Also like to deprecate `pkt["IP"]`, but that’s another story)

Reply:
@p-l- @guedou so what are your calls ?

Reply:
I find it useful and I think that i can be merged. However, I still wonder if the problem that it is adressing (using Dot11FCS instead of Dot11) is indeed a regression.

Reply:
I have a problem with the current implementation.

I don't think (Ether()/Dot11FCS()/IP())[Packet] really "means" something anyway. Nobody would use it.

Another option would be to replace the check with an `isinstance()` for all the layers, so that `(Ether()/Dot11FCS()/IP())[Packet]` returns the `Ether` layer.

That would make more sense **but** we have to be careful as it changes the behaviour (this PR changes current the behaviour anyway), and we want to make sure that every inheritance we have for "coding" reasons is also meaningful from a "protocol" point of view (e.g., if `A` inherits from `B` because both protocols are similar, but they A is not a "more specific version" of B, we should create `_A_and_B`, and have both `A` and `B` inherit from it.

What do you think?

Reply:
The change of behavior is avoided by the fact that we introduce a new class :/

I don't really get what you'd rather have: feel free to submit a PR with a similar fix to https://github.com/secdev/scapy/issues/1590 if you have an idea :)

Reply:
What I meant is: maybe we could, actually, change the current behaviour, and simply use `isinstance()` everywhere?